lib/oli/grading.ex

```diff
@@ -211,6 +211,44 @@ defmodule Oli.Grading do
     {gradebook, column_labels}
   end
 
+  @doc """
+  Returns a list of GradebookScore for enrolled user in the provided section
+
+  `[%GradebookScore{}, GradebookScore{}, ...]`
+  """
+  def get_scores_for_section_and_user(%Section{} = section, student_id) do
+    # get publication page resources, filtered by graded: true
+    graded_pages = Sections.fetch_scored_pages(section.slug)
+
```

Comment:
This should be optimized one step further.  

Here is what is taking place with this latest change:
1. Query for all scored_pages
2. Query for all resource accesses (for all students)
3. Post process to filter down to only those resource accesses for this specific student (and convert to GradebookScore)

Instead, the two queries can be merged into one in a way that eliminates the need for any post processing:

Something like: 

```
SELECT page.resource_id, page.title, resource_access.score, resource_access.out_of, resource_access.was_late
FROM subquery_for_graded_page_titles_and_resource_id_only(section.id) as page
LEFT JOIN resource_accesses resource_access ON resource_access.id = page.resource_access_id
WHERE resource_access.section_id = ^section.id and resource_access.user_id = ^user.id
```

The above isn't tested and likely not valid SQL, but it conveys the intent of what we want here - a single query that gets just the information that we need, which can simply then be mapped into the `GradebookScore` structs.  Also, this SQL approach will right-pad with `nil` resource access records that are missing.  A "missing" resource_access record means that the student has never visited that particular page yet.  In that case, we still want to list the quiz in the table, with some indication that the student has no score yet.  The existing approach I believe will omit graded pages that student has not visited (given how it reduces only non `nil` values into the final list)